tools/data-api-sdk/v1/models/terraform_schema_object_definition_type.go

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package models
+
+// TerraformSchemaObjectDefinitionType defines the Type of Object Definition used in an TerraformSchemaObjectDefinition.
+// Possible values include Simple types (boolean/integer/float/string), References (to a Constant or Model) and a
+// List of the previous values.
+type TerraformSchemaObjectDefinitionType = string
+
+// NOTE: we intentionally only have Terraform Schema fields (and specific CustomSchema types) here - meaning
+// that we don't have RawObject/RawFile since we have no means of expressing them today.
+
+// Simple Types
+
+const (
+	// BooleanTerraformSchemaObjectDefinitionType specifies that this represents a Boolean value.
+	// This is output as a `bool` in Models and as `pluginsdk.TypeBool` within the Terraform Schema.
+	BooleanTerraformSchemaObjectDefinitionType TerraformSchemaObjectDefinitionType = "Boolean"
```

Comment:
In short: mostly. Longer answer:

Today we're combine 2 of the 3 Object Definitions together (which is actually problematic since it means we're not correctly modelling the `option` fields correctly)  - so this PR now intentionally splits the Object Definition 3 ways rather than 2 ways, into:

1. `SDKObjectDefinition` - used to represent Golang types (used in the Go SDK).
2. `OperationOptionObjectDefinition` - used to represent types used in the QueryString/Headers (used in the Go SDK). These are Golang types but only support a subset of values (bool/int/float/string/list/csv/constant).
3. `TerraformSchemaObjectDefinition` - used to represent the types used in Terraform, which aren't 1:1 with the `SDKObjectDefinition` (used in the Terraform Generator).

to ensure we're representing those correctly - but all of those models come from the new `data-api-sdk` package.

Whilst we'll be using those models throughout `importer-rest-api-specs` (and `importer-msgraph-metadata` soon) - we'll need to use a different model for persisting those to disk (since not all of the API Definitions are stored as JSON [e.g. the test configurations are HCL] - so we'll ned to transform to/from that) - but we should be able to remove the duplicated types from `importer-rest-api-specs`, yes.